File: parser/src/main/java/io/ticofab/androidgpxparser/parser/domain/WayPoint.java
package io.ticofab.androidgpxparser.parser.domain;

/**
 * A way point (wpt element).
 */
public class WayPoint extends Point {

    private WayPoint(Builder builder) {
        super(builder);
    }

    public static class Builder extends Point.Builder {

        @Override
        public WayPoint build() {
            return new WayPoint(this);
        }
    }

}

File: parser/src/main/kotlin/io/ticofab/androidgpxparser/parser/domain/WayPoint.kt
package io.ticofab.androidgpxparser.parser.domain

/**
 * A way point (wpt element).
 */
class WayPoint private constructor(builder: Builder) : Point(builder) {
    class Builder : Point.Builder() {
        override fun build() = WayPoint(this)
    }
}